pennylane/ops/qubit.py

```diff
@@ -515,10 +606,15 @@ class PhaseShift(Operation):
     grad_method = "A"
     generator = [np.array([[0, 0], [0, 1]]), 1]
 
-    @staticmethod
-    def _matrix(*params):
+    @classmethod
+    def _matrix(cls, *params):
```

Comment:
Just curious (maybe I haven't spotted it yet), but is there a specific reason `_matrix` was made a class method?

Reply:
The idea behind it was that for `DiagonalOperation` you could implement it with `np.diag(cls._eigvals())`.

Reply:
And if you want to overwrite the method in subclasses, a classmethod appears more sensible to me